Mathlib/Analysis/Normed/Field/ProperSpace.lean

```diff
@@ -0,0 +1,46 @@
+/-
+Copyright (c) 2024 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+
+import Mathlib.Analysis.Normed.Field.Lemmas
+import Mathlib.Analysis.SpecificLimits.Basic
+import Mathlib.Topology.MetricSpace.ProperSpace
+
+/-!
+# Proper nontrivally normed fields
+
+Nontrivially normed fields are `ProperSpaces` when they are `WeaklyLocallyCompact`.
+
+## Main results
+
+* `ProperSpace.of_nontriviallyNormedField_of_weaklyLocallyCompactSpace`
+
+## Implementation details
+
+This is a special case of `ProperSpace.of_locallyCompactSpace` from
+`Mathlib.Analysis.Normed.Module.FiniteDimension`, specialized to be on the field itself
+with a proof that requires fewer imports.
```

Comment:
You could (entirely optional) add some `assert_not_exists` statements here that would defend against import creep. Only if it makes sense.